cpp/daal/include/data_management/data/internal/numeric_table_sycl_soa.h

```diff
@@ -598,41 +563,62 @@ class DAAL_EXPORT SyclSOANumericTable : public SyclNumericTable
                 TypeDispatcher::dispatch(uniBuffer.type(), converter);
 
                 _arrays[j] = converter.getResult(st);
-                DAAL_CHECK_STATUS_VAR(st);
+                if (!st) return services::throwIfPossible(st);
             }
         }
         block.reset();
         return services::Status();
     }
 
     template <typename T>
-    services::Status getTFeature(size_t feat_idx, size_t idx, size_t nrows, int rwFlag, BlockDescriptor<T> & block)
+    services::Status getTFeature(size_t feat_idx, size_t idx, size_t nrows, ReadWriteMode rwFlag, BlockDescriptor<T> & block)
     {
         using namespace services::internal::sycl;
 
-        const size_t nobs = getNumberOfRows();
+        if (isCpuTable())
+        {
+            return _cpuTable->getBlockOfColumnValues(feat_idx, idx, nrows, rwFlag, block);
+        }
+
+        const size_t nobs  = getNumberOfRows();
+        const size_t ncols = getNumberOfColumns();
+
+        if (feat_idx >= ncols)
+        {
+            return services::throwIfPossible(services::ErrorIncorrectIndex);
+        }
+
         block.setDetails(feat_idx, idx, rwFlag);
 
         if (idx >= nobs)
         {
-            block.resizeBuffer(1, 0);
-            return services::Status(services::ErrorIncorrectIndex);
+            if (!block.resizeBuffer(1, 0))
+            {
+                return services::throwIfPossible(services::ErrorMethodNotSupported);
```

Comment:
Maybe ErrorMemoryAllocationFailed?

Reply:
Actually `resizeBuffer` in this case simply deallocates the memory - so ErrorMemoryAllocationFailed midth confuse. Also, when deallocating, it always returns true - but I check the return status in case if the logic of this method will be changed.